extensions-contrib/dropwizard-emitter/src/main/java/org/apache/druid/emitter/dropwizard/DropwizardEmitter.java

```diff
@@ -0,0 +1,171 @@
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.core.Emitter;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.service.AlertEvent;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DropwizardEmitter implements Emitter
+{
+  private static final Logger log = new Logger(DropwizardEmitter.class);
+  private final MetricRegistry metricsRegistry = new MetricRegistry();
+  private final AtomicBoolean started = new AtomicBoolean(false);
+  private final DropwizardConverter converter;
+  private final List<Emitter> alertEmitters;
+  private final List<DropwizardReporter> reporters;
+  private final DropwizardEmitterConfig config;
+  // Note: the gauges do not represent the actual instantaneous value for the metrics.
+  // Instead they have the last known value for it.
+  private final Map<String, Number> gagues;
+
+  public DropwizardEmitter(
+      DropwizardEmitterConfig config,
+      ObjectMapper mapper,
+      List<Emitter> alertEmitters
+  )
+  {
+    this.alertEmitters = alertEmitters;
+    this.config = config;
+    this.reporters = config.getReporters();
+    this.converter = new DropwizardConverter(mapper, config.getDimensionMapPath());
+    this.gagues = Collections.synchronizedMap(new GaugesCache<>(config.getMaxGaugeCount()));
+  }
+
+
+  @Override
+  public void start()
+  {
+    final boolean alreadyStarted = started.getAndSet(true);
+    if (!alreadyStarted) {
+      for (DropwizardReporter reporter : reporters) {
+        reporter.start(metricsRegistry);
+      }
+    }
+  }
+
+  @Override
+  public void emit(Event event)
+  {
+    synchronized (started) {
+      if (!started.get()) {
+        throw new RejectedExecutionException("Dropwizard emitter Service not started.");
+      }
+    }
+    if (event instanceof ServiceMetricEvent) {
+      ServiceMetricEvent metricEvent = (ServiceMetricEvent) event;
+      String host = metricEvent.getHost();
+      String service = metricEvent.getService();
+      String metric = metricEvent.getMetric();
+      Map<String, Object> userDims = metricEvent.getUserDims();
+      Number value = metricEvent.getValue();
+      ImmutableList.Builder<String> nameBuilder = new ImmutableList.Builder<>();
+      LinkedHashMap<String, String> dims = new LinkedHashMap<>();
+      final DropwizardMetricSpec metricSpec = converter.addFilteredUserDims(service, metric, userDims, dims);
+
+      if (metricSpec != null) {
+        if (config.getPrefix() != null) {
+          nameBuilder.add(config.getPrefix());
+        }
+        nameBuilder.add("metric=" + metric);
+        nameBuilder.add("service=" + service);
```

Comment:
should use string format instead of concatenations here and in all other places as this code is gonna be called many many times.